### PR TITLE
eval(m, expr) -> Core.eval(m, expr)

### PR DESCRIPTION
**Pull request by fredrikekre**
fixes incoming deprecations from https://github.com/JuliaLang/julia/pull/27098

### CODE REVIEW

File: src/DocChecks.jl [hunk 1]
```diff
@@ -144,8 +144,7 @@ function doctest(block::Markdown.Code, meta::Dict, doc::Documents.Document, page
         sym = isempty(name) ? gensym("doctest-") : Symbol("doctest-", name)
         sandbox = get!(page.globals.meta, sym) do
             newmod = Module(sym)
-            eval(newmod, :(eval(x) = Core.eval($newmod, x)))
-            eval(newmod, :(eval(m, x) = Core.eval(m, x)))
+            Core.eval(newmod, :(eval(x) = Core.eval($newmod, x)))
```
Comment (mortenpi) [inline src/DocChecks.jl]: The second one is not necessary anymore? Why do we need to do this here in the first place?

**Comment by mortenpi** [comment]
Will release 0.18 when this and #718 are in.

File: src/DocChecks.jl (hunk 1 above)
Comment (fredrikekre) [reply src/DocChecks.jl]: It is needed for things like
````
```jldoctest
julia> a = 1; b = 2;

julia> eval(:(a + b))
3
```
````
similar to https://github.com/JuliaLang/julia/pull/27098/files#diff-8d2b8a22d475d81e3a1944aedb57ad42R391

Although, the docstring for single arg `eval` states:

> Evaluate an expression in the current module. Every `Module` (except those defined
with `baremodule`) has its own 1-argument definition of `eval`, which evaluates
expressions in that module.

So presumably the sandbox module should already have its own `eval` defined, which should evaluate into the sandbox, but I get
```
┌ Warning: `eval(e)` is deprecated, use `Core.eval(Main, e)` instead.
│   caller = top-level scope
└ @ Core :0
ERROR: UndefVarError: a not defined
Stacktrace:
 [1] top-level scope
 [2] eval at ./boot.jl:316 [inlined]
 [3] eval(::Expr) at ./deprecated.jl:55
 [4] top-level scope

```
on your branch @JeffBezanson.

Comment (mortenpi) [reply src/DocChecks.jl]: Might be good to add a small comment there to say what the `eval`s do, for future reference.

**Pull request merged**